ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.ts

```diff
@@ -27,20 +37,81 @@ export class HyperdriveJobComponent implements OnInit, OnDestroy {
   @Input() jobParameters: HyperdriveDefinitionParametersModel;
   @Output() jobParametersChange: EventEmitter<HyperdriveDefinitionParametersModel> = new EventEmitter();
   @Input() isJobTemplateSelected: boolean;
-  @Input() jobTemplateChanges: EventEmitter<string>;
+  @Input() jobTemplateChanges: EventEmitter<JobTemplateChangeEventModel>;
+  @Input() jobTemplateParameters: HyperdriveTemplateParametersModel;
+
+  private hyperdriveType: string = undefined;
 
   jobTemplateChangesSubscription: Subscription;
 
+  hyperdriveTypes = hyperdriveTypes;
+  hyperdriveTypesMap = hyperdriveTypesMap;
+  hyperdriveFields = hyperdriveFields;
+
   constructor() {
     // do nothing
   }
 
   ngOnInit(): void {
-    this.jobTemplateChangesSubscription = this.jobTemplateChanges.subscribe((value) => {
-      if (value) {
-        this.jobParametersChange.emit({ ...this.jobParameters, jobJar: undefined, mainClass: undefined });
+    this.jobTemplateChangesSubscription = this.jobTemplateChanges.subscribe((value: JobTemplateChangeEventModel) => {
+      this.jobParametersChange.emit({
+        ...this.jobParameters,
+        jobJar: undefined,
+        mainClass: undefined,
+        appArguments: this.getAppArguments(),
+      });
+      if (value?.jobTemplateId) {
+        const jobTemplateParameters = <HyperdriveTemplateParametersModel>value?.jobTemplateParameters;
+        this.hyperdriveType = HyperdriveUtil.getHyperdriveTypeFromAppArguments(
+          jobTemplateParameters.appArguments,
+          hyperdriveTypesJobTemplateFields,
+        );
       }
     });
+
+    if (this.isJobTemplateSelected) {
```

Comment:
I would extract this block to a separate function `initHyperdriveType` or something like that

Reply:
Done